packages/blade/src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { DropdownContext, useDropdown } from './useDropdown';
+import type { DropdownContextType } from './useDropdown';
+import { useId } from '~src/hooks/useId';
+import Box from '~components/Box';
+import { getPlatformType } from '~utils';
+
+type DropdownProps = {
+  selectionType?: 'single' | 'multiple';
+  children: React.ReactNode[];
+};
+
+const platformType = getPlatformType();
+const isReactNative = platformType === 'react-native';
+
+/**
+ * **Dropdown component**
+ *
+ * **Usage**
+ *
+ * ```jsx
+ * <Dropdown selectionType="single">
+ *  <SelectInput />
+ *  <DropdownOverlay>
+ *    <ActionList>
+ *      <ActionListItem />
+ *      <ActionListItem />
+ *    </ActionList>
+ *  </DropdownOverlay>
+ * </Dropdown>
+ * ```
+ */
+function Dropdown({ children, selectionType = 'single' }: DropdownProps): JSX.Element {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [options, setOptions] = React.useState<DropdownContextType['options']>([]);
+  const [selectedIndices, setSelectedIndices] = React.useState<
+    DropdownContextType['selectedIndices']
+  >([]);
+  const [activeIndex, setActiveIndex] = React.useState(-1);
+  const [shouldIgnoreBlur, setShouldIgnoreBlur] = React.useState(false);
+  const selectInputRef = React.useRef<HTMLButtonElement>(null);
+  const actionListRef = React.useRef<HTMLDivElement>(null);
+
+  const dropdownBaseId = useId('dropdown');
+
+  return (
+    <DropdownContext.Provider
+      value={{
+        isOpen,
+        setIsOpen,
+        selectedIndices,
+        setSelectedIndices,
+        options,
+        setOptions,
+        activeIndex,
+        setActiveIndex,
+        shouldIgnoreBlur,
```

Comment:
We ignore the blur for certain cases like when user is clicking on the option or when user is focusing on Apply button on footer.

Reply:
> we won't ever need this to be tab navigated? the tab navigation will be on the action list always?

Not sure why reply option wasn't showing there so replying here-

It was unnecessary. In #966 , I removed it from here and added it to ActionListItem because we want to make it tabbable for cases with Footer.